TeamProject/GlobalWarmingGame/Interactions/Interactables/Buildings/WorkBench.cs

```diff
@@ -61,165 +61,57 @@ public class WorkBench : Sprite, IInteractable, IBuildable, IUpdatable
-            }
-            //follower.Goals.Clear();
+            WorkBenchCrafter(follower, Craftable.bow);
         }
 
         private void CraftHoe(IInstructionFollower follower)
         {
-            CraftableType hoe = ResourceTypeFactory.MakeCraftable(Craftable.hoe);
-            if (follower.Inventory.CheckContainsList(hoe.CraftingCosts))
-            {
-                foreach (ResourceItem item in hoe.CraftingCosts)
-                {
-                    follower.Inventory.RemoveItem(item);
-                    Console.WriteLine("Removed " + item.Type.DisplayName + " amount: " + item.Amount);
-                }
-                Console.WriteLine("Added hoe" + " amount: " + 1);
-                follower.Inventory.AddItem(new ResourceItem(hoe, 1));
-            }
-
-            else
-            {
-                ResourceNotification.Visible = true;
-            }
-            //follower.Goals.Clear();
+            WorkBenchCrafter(follower, Craftable.hoe);
         }
 
         private void CraftPickaxe(IInstructionFollower follower)
         {
-            CraftableType pickaxe = ResourceTypeFactory.MakeCraftable(Craftable.pickaxe);
-            if (follower.Inventory.CheckContainsList(pickaxe.CraftingCosts))
-            {
-                foreach (ResourceItem item in pickaxe.CraftingCosts)
-                {
-                    follower.Inventory.RemoveItem(item);
-                    Console.WriteLine("Removed " + item.Type.DisplayName + " amount: " + item.Amount);
-                }
-                Console.WriteLine("Added pickaxe" + " amount: " + 1);
-                follower.Inventory.AddItem(new ResourceItem(pickaxe, 1));
-            }
-
-            else
-            {
-                ResourceNotification.Visible = true;
-            }
-            //follower.Goals.Clear();
+            WorkBenchCrafter(follower, Craftable.pickaxe);
         }
 
         private void CraftBasicRifle(IInstructionFollower follower)
         {
-            CraftableType basicRifle = ResourceTypeFactory.MakeCraftable(Craftable.basicRifle);
-            if (follower.Inventory.CheckContainsList(basicRifle.CraftingCosts))
+            WorkBenchCrafter(follower, Craftable.basicRifle);
+        }
+
+        private void WorkBenchCrafter(IInstructionFollower follower, Craftable craftableEnum)
+        {
+            CraftableType craftable = ResourceTypeFactory.MakeCraftable(craftableEnum);
+            if (follower.Inventory.CheckContainsList(craftable.CraftingCosts))
             {
-                foreach (ResourceItem item in basicRifle.CraftingCosts)
+                foreach (ResourceItem item in craftable.CraftingCosts)
                 {
                     follower.Inventory.RemoveItem(item);
                     Console.WriteLine("Removed " + item.Type.DisplayName + " amount: " + item.Amount);
                 }
                 Console.WriteLine("Added BasicRifle" + " amount: " + 1);
```

Comment:
Looks good and all the functionality is there, but on line 112, it is still printing "Added BasicRifle" to the console. This should print the name of the craftable. Perhapse this line could be removed entirely, it still might be useful for when we redo the inventory ui. Perhapse we should think about a more scalable logger solution.